internal/internal_workflow_testsuite.go

```diff
@@ -684,6 +697,35 @@ func (env *testWorkflowEnvironmentImpl) Complete(result []byte, err error) {
 	}
 }
 
+func (h *testWorkflowHandle) retryAsChild() bool {
+	env := h.env
+	if !env.isChildWorkflow() {
+		return false
+	}
+	params := h.params
+	if params.retryPolicy != nil && env.testError != nil {
+		errReason, _ := getErrorDetails(env.testError, env.GetDataConverter())
+		var expireTime time.Time
+		if params.retryPolicy.GetExpirationIntervalInSeconds() > 0 {
+			expireTime = params.scheduledTime.Add(time.Second * time.Duration(params.retryPolicy.GetExpirationIntervalInSeconds()))
+		}
+		backoff := getRetryBackoffFromThriftRetryPolicy(params.retryPolicy, env.workflowInfo.Attempt, errReason, env.Now(), expireTime)
+		if backoff > 0 {
+			env.registerDelayedCallback(func() {
+				// remove the current child workflow from the pending child workflow map because
+				// the childWorkflowID will be the same for retry run.
+				delete(env.runningWorkflows, env.workflowInfo.WorkflowExecution.ID)
```

Comment:
What's going to happen if signal is sent to a child while waiting for a retry? I think in real life it should be delivered to the next run.

Reply:
Just realize that currently, there is no way to send signal to child workflow in the unit test environment. https://github.com/uber-go/cadence-client/issues/564 will address that separately.